### PR TITLE
feat: Prevent deletes in consumers and streams

### PR DESCRIPTION
**Pull request by danielcibrao-form3**
Adding a `preventDelete` property for streams and consumers which when false
nack should be able to delete it, otherwise it should just ignore that
stream or consumer.

The use case for this is in a multi-cloud context when we have deploy
deployed in different clouds it becomes troublesome. For example, nack
on GKE might check that a stream CRD is missing and proceeds to delete
it, however that CRD exists in AWS and AKS and it should not be deleted
because its going to impact other applications.

### CODE REVIEW

**Comment by caleblloyd** [comment]
Managed Resource Deletion has historically been handled in Finalizers in Operators, but in NACK is being handled in a list/clean loop that was implemented in #56 

I have opened #90 to try to figure out if we should go back to a Finalizer approach for resource deletion

If we used finalizers, then only a Deleted CRD Finalizer would be able to delete a Managed NATS Stream or Consumer, so you wouldn't get any ghost deletes from a missing CRD

File: deploy/crds.yml [hunk 1]
```diff
@@ -189,6 +189,10 @@ spec:
                   source:
                     type: string
                     description: Messages will be published from that subject to the destination subject.
+              preventDelete:
+                description: Indicates if a Stream can be deleted (true) or not (false)
```
Comment (caleblloyd) [inline deploy/crds.yml:193]: This should say `When true, the managed Stream will not be deleted when the resource is deleted`

File: deploy/crds.yml [hunk 2]
```diff
@@ -551,6 +555,10 @@ spec:
                 description: Name of the account to which the Consumer belongs.
                 type: string
                 pattern: '^[^.*>]*$'
+              preventDelete:
+                description: Indicates if a Consumer can be deleted (true) or not (false)
```
Comment (caleblloyd) [inline deploy/crds.yml:559]: This should say `When true, the managed Consumer will not be deleted when the resource is deleted`

**Review by caleblloyd — CHANGES_REQUESTED** [review]
Wording in the descriptions is backwards, I've suggested some wording that should be a little clearer

File: deploy/crds.yml (hunk 2 above)
Comment (caleblloyd) [reply deploy/crds.yml]: I made this change to the description

File: deploy/crds.yml (hunk 1 above)
Comment (caleblloyd) [reply deploy/crds.yml]: I made this change to the description

**Review by caleblloyd — COMMENTED** [review]
I updated the descriptions, I think this looks good now.  Would like to get @wallyqs opinion on it too though

**Review by wallyqs — APPROVED** [review]
LGTM

**Review by caleblloyd — APPROVED** [review]

**Pull request merged**